src/Plugin/AddPathPlugin.php

```diff
@@ -39,9 +46,14 @@ public function __construct(UriInterface $uri)
      */
     public function handleRequest(RequestInterface $request, callable $next, callable $first)
     {
-        $request = $request->withUri($request->getUri()
-            ->withPath($this->uri->getPath().$request->getUri()->getPath())
-        );
+        $identifier = spl_object_hash((object) $first);
+
+        if (!in_array($identifier, $this->alteredRequests)) {
```

Comment:
could we make alteredRequests a hashmap of `$identifier => true` instead? then we could check with `array_has_key` instead of `in_array`. for long running processes that send lots of requests, the `in_array` will become expensive once the list of altered requests grows.

Reply:
That came in my mind but IMO the chance of this is really low. Anyway, I updated to code to store `identifier => identifier` in the array therefore we do not need to check what is the value of an identifier. Storing boolean values in the array would be also misleading because someone may think that `$identifier => false` means the request has not been altered yet.